UnitsNet/CustomCode/UnitAbbreviationsCache.cs

```diff
@@ -90,6 +90,26 @@ private void LoadGeneratedAbbreviations()
             MapUnitToAbbreviation(typeof(TUnitType), Convert.ToInt32(unit), GlobalConfiguration.DefaultCulture, abbreviations);
         }
 
+        /// <summary>
+        /// Adds a unit abbreviation for the given unit enum value and sets it as the default.
+        /// This is used to dynamically add abbreviations for existing unit enums such as <see cref="UnitsNet.Units.LengthUnit"/> or to extend with third-party unit enums
+        /// in order to <see cref="UnitParser.Parse{TUnitType}"/> or <see cref="GetDefaultAbbreviation{TUnitType}"/> on them later.
+        /// </summary>
+        /// <param name="unit">The unit enum value.</param>
+        /// <param name="abbreviation">Unit abbreviations to add as default.</param>
+        /// <typeparam name="TUnitType">The type of unit enum.</typeparam>
+        [PublicAPI]
+        // Windows Runtime Component does not allow public methods/ctors with same number of parameters: https://msdn.microsoft.com/en-us/library/br230301.aspx#Overloaded methods
+#if WINDOWS_UWP
+        internal
+#else
+        public
+#endif
+            void MapUnitToDefaultAbbreviation<TUnitType>(TUnitType unit, string abbreviation) where TUnitType : Enum
```

Comment:
Can we maybe add a `bool setAsDefault = false` parameter to the existing `MapUnitToAbbreviation` methods instead? Windows Runtime Component should be fine since it's already `internal` for that target.

Reply:
I thought of that, but adding default values to `params` methods is gross.  Also, I only want a single default.

Reply:
Ah I forgot it had a `params`, then I agree. 

But it seems we have a bit of duplicate code in the new and old methods, can we maybe:
1. Extract common code to a private method
2. Rename `MapUnitToDefaultAbbrevation` to `SetDefaultAbbreviationForUnit`, I'm not too fond of the old naming scheme
3. Add unit tests for the new methods, see how we test the old methods (I think we have some)

Reply:
Sure.  I'll put some of that together and update.

Reply:
✔️ see https://github.com/angularsen/UnitsNet/pull/578/files#diff-96c7d0bea2108a11d25f686fd1687acaR209

Reply:
Um... regarding the tests... it seems that only a few of the generated test base classes are being produced.  I'm missing the vast majority of them.

Reply:
Nevermind.  Found the batch file.  It's really named quite well. 😄

Reply:
I still like the "map" word here since it's still technically mapping, if the abbreviation doesn't exist.  But it's your library. If you really want "set," I'll change it to "set."

Reply:
I see your point and I can go either way. It's not a big deal for me and we can keep it as-is. 
My biggest gripe with the naming is that it seems backwards to me. In my mind, we don't map a unit to an abbreviation, we map abbrevation(s) to a unit. But can't change the old method name now, so let's just go with it for now. It's fine.